src/Host/Storage.php

```diff
@@ -14,6 +14,73 @@
 
 class Storage
 {
+	/**
+	 * In a multi user environment where multiple projects are located on the same instance(s)
+	 *	if two or more users are deploying at the same time on the same environment, 
+	 *	the generated hosts files are overwritten or worse, the deploy will fail 
+	 *	due to file permissions
+	 *
+	 * Make the persistent storage folder configurable or use the system tmp 
+	 *	as default if not possible
+	 * 
+	 * @return string
+	 * @throws Exception
+	 */
+	private static function _getPersistentStorageLocation() {
+		$config = \Deployer\Deployer::get()->config;
+
+		// use the system temporary folder and the current pid as default 
+		// persistent storage in case we can't use the configured value
+		// or we can't create the default location
+		$tmp = sys_get_temp_dir() . '/' . posix_getpid();
+
+		// get the location for the deployer configuration
+		$deployerConfig = $config->has('deployer_config') ? $config->get('deployer_config') : null;
+
+		if ( !is_null($deployerConfig) ) {
+			// check if the folder exists and it's writable
+			if ( !(is_dir($deployerConfig) && is_writable($deployerConfig)) ) {
+				throw new Exception("Deployer folder `$deployerConfig` doesn't exists or doesn't writable.");
+			}
+		} else {
+			// not configured, generate deployer folder name
+
+			// use the home dir of the current user 
+			// and the repository name
+			$userInfo = posix_getpwuid(posix_getuid());
```

Comment:
For non posix compatible i'm going to use getenv.
getenv('HOMEDRIVE'), getenv('HOMEPATH') are available on windows 8+ 
getenv('HOME') for *nix

It's Windows 8+ enough?